File: Python3/2379. Minimum Recolors to Get K Consecutive Black Blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        l = white = 0
        ans = k

        for i, c in enumerate(blocks):
            if c == 'W':
                white += 1

            if i - l + 1 == k:
                ans = min(ans, white)
                white -= blocks[l] == 'W'
                l += 1

        return ans

File: Java/2379. Minimum Recolors to Get K Consecutive Black Blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int l = 0, white = 0, ans = k;

        for (int r = 0; r < blocks.length(); r++) {
            char c = blocks.charAt(r);
            if (c == 'W') {
                white++;
            }

            if (r - l + 1 == k) {
                ans = Math.min(ans, white);
                white -= blocks.charAt(l) == 'W' ? 1 : 0;
                l++;
            }
        }
        
        return ans;
    }
}